### DISCUSSION TITLE
ReactCurrentOwner

### DISCUSSION THREAD

**Discussion by SenneChristiaens** [vercel/next.js#73523] [Help] [answered]
### Summary

I installed three and fiber. When I create a Canvas I immediately get this error:
![image](https://github.com/user-attachments/assets/582300a9-50f9-4651-903c-d0cd70579432)

This is my package.json:

```
{
  "name": "senne-christiaens-portfolio",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "slicemachine": "start-slicemachine"
  },
  "dependencies": {
    "@gsap/react": "^2.1.1",
    "@prismicio/client": "^7.12.0",
    "@prismicio/next": "^1.7.1",
    "@prismicio/react": "^2.9.1",
    "@types/three": "^0.170.0",
    "clsx": "^2.1.1",
    "gsap": "^3.12.5",
    "next": "15.0.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "@react-three/fiber": "^8.15.12",
    "@react-three/drei": "^9.92.1",
    "three": "^0.159.0"
  },
  "devDependencies": {
    "@slicemachine/adapter-next": "^0.3.58",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "autoprefixer": "^10.4.20",
    "eslint": "^8",
    "eslint-config-next": "15.0.3",
    "postcss": "^8",
    "slice-machine-ui": "^2.10.4",
    "tailwindcss": "^3.4.1",
    "typescript": "^5"
  }
}
```

This is my Canvas:

```
"use client";

import React from "react";
// import * as THREE from "three";
import { Canvas } from "@react-three/fiber";
import { ContactShadows, Float, Environment } from "@react-three/drei";
import { Suspense, useRef, useEffect, useState } from "react";
// import { gsap } from "gsap";
// import { useGSAP } from "@gsap/react";

export function Shapes() {
  return (
    <div className="row-span-1 row-start-1 -mt-9 aspect-square  md:col-span-1 md:col-start-2 md:mt-0">
      <Canvas
        className="z-0"
        shadows
        gl={{ antialias: false }}
        dpr={[1, 1.5]}
        camera={{ position: [0, 0, 25], fov: 30, near: 1, far: 40 }}
      >
        <Suspense fallback={null}>
          <ContactShadows
            position={[0, -3.5, 0]}
            opacity={0.65}
            scale={40}
            blur={1}
            far={9}
          />
          <Environment preset="studio" />
        </Suspense>
      </Canvas>
    </div>
  );
}
```

Does anyone have a solution to my problem?

**Comment by icyJoseph** [comment]
For now, you need to use,`@react-three/fiber@alpha`, as pointed out here:

- https://github.com/pmndrs/react-three-fiber/issues/3398#issuecomment-2494653767:

And further developed here, https://github.com/pmndrs/react-three-fiber/issues/3222

**Reply by SenneChristiaens** [reply]
I've tried that, got this:
![image](https://github.com/user-attachments/assets/ffe0c772-ff67-49d1-8109-4efaaccdc20a)

I've tested every version of React, the 19 RC one, 18.0.0.... nothing seems to work....

**Reply by icyJoseph** [reply]
Well that's the solution, you've just hit another issue, which is the dependencies compatibility. 

There's another discussion around where people are sharing their solutions.

Later I can take a look, otherwise. 

Did you try to install React 19 RC? What next.js version do you have? 

Got reproduction repository?

**Reply by icyJoseph** [reply]
Apparently there's a beta version too, https://github.com/pmndrs/react-three-fiber/issues/3222#issuecomment-2443940258

You might have to force the dependency with, -- force, because rc doesn't seem to fulfil greater than 19... Odd

**Reply by SenneChristiaens** [reply]
I am using 15.0.3. I've tried with 19RC because I've created my app with create-next-app@latest. So I downgraded to a stable version but that didn't solve the issue. I've tried all combinations. 19RC with and without alpha as well as 18.2.0 with and without alpha...

**Reply by SenneChristiaens** [reply]
If I force it, everything crashes.

**Reply by icyJoseph** [reply]
Mmm I'll give a go when I get home. Too bad the issue is upstream in react-three-fiber... Not that much can be done here

**Reply by SenneChristiaens** [reply]
Feel free to take a look: https://github.com/SenneChristiaens/senne-christiaens-portfolio

The Canvas is located in: "/src/slices/Hero/Shapes.jsx"
And imported at: "src/slices/Hero/index.tsx"

I commented everything out and moved on. Thanks in advance!

**Reply by icyJoseph** [reply]
React 19 is stable and out now! Could you give it a go?

**Reply by SenneChristiaens** [reply]
So upgrade back to the rc version? And try with the alpha version?

**Reply by icyJoseph** [reply]
Try with react-fiber beta or alpha, and install React 19.0.0, released some minutes ago

**Reply by icyJoseph** [reply, marked as answer]
Ok, I got it to install without dependency conflicts. Needed to do some overrides, and I was able to install `@react-three/fiber@beta`:

```json
  "overrides": {
    "@gsap/react": {
      "react": "19.0.0"
    },
    "@prismicio/next": {
      "react": "19.0.0",
      "next": "15.0.3"
    },
    "next": {
      "react": "19.0.0",
      "react-dom": "19.0.0"
    },
    "@react-three/drei": {
      "@react-three/fiber": "9.0.0-beta.1"
    }
  }
```

My suggestion, wait a bit for Next, and others to add React 19 to their peerDependencies, about an hour ago Next.js updated it, and it is most likely coming out on the next release. Same with other packages that depended on specific RC versions.

You can see a running version here, https://stackblitz.com/edit/github-ex91vh?file=package.json

**Reply by SenneChristiaens** [reply]
Thanks Joseph, Hero!!

**Reply by icyJoseph** [reply]
If it's all working fine, somewhat, please don't forget to mark as answered 😄
